features/nfc/source/nfc/NFCEEPROM.cpp

```diff
@@ -0,0 +1,409 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "NFCEEPROM.h"
+#include "ndef/ndef.h"
+
+using namespace mbed;
+using namespace mbed::nfc;
+
+NFCEEPROM::NFCEEPROM(NFCEEPROMDriver *driver, events::EventQueue *queue, const Span<uint8_t> &ndef_buffer) : NFCTarget(ndef_buffer),
+    _delegate(NULL), _driver(driver), _initialized(false), _current_op(nfc_eeprom_idle), _ndef_buffer_read_sz(0), _eeprom_address(0), _operation_result(NFC_ERR_UNKNOWN)
+{
+    _driver->set_delegate(this);
+    _driver->set_event_queue(queue);
+}
+
+nfc_err_t NFCEEPROM::initialize()
+{
+    MBED_ASSERT(_initialized == false); // Initialize should only be called once
+
+    // Initialize driver
+    _driver->reset();
+    _initialized = true;
+    return NFC_OK;
+}
+
+void NFCEEPROM::set_delegate(NFCEEPROM::Delegate *delegate)
+{
+    _delegate = delegate;
+    set_ndef_delegate(delegate);
```

Comment:
why is this stored twice?

Reply:
`NFCNDEFCapable` has its own delegate-handling system